salt/modules/chroot.py

```diff
@@ -347,7 +347,9 @@ def sls(root, mods, saltenv="base", test=None, exclude=None, **kwargs):
         high_data = st_.state.apply_exclude(high_data)
 
         # Compile and verify the raw chunks
-        chunks = st_.state.compile_high_data(high_data)
+        chunks, errors = st_.state.compile_high_data(high_data)
+        if errors:
+            return errors
```

Comment:
note: If this is an error condition, it should set the retcode accordingly.

```suggestion
        if errors:
            __context__["retcode"] = salt.defaults.exitcodes.EX_STATE_COMPILER_ERROR
            return errors
```

Reply:
Good suggestions/finds. Should this retcode also be set in all the existing `if errors:` lines above too then like on line 339 and 344?

Reply:
Yes. Weird, `salt.defaults.exitcodes` is even imported, but never used in this module.